src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1032,7 +1032,7 @@ public bool TakeStartLine(ReadableBuffer buffer, out ReadCursor consumed, out Re
             var result = _parser.ParseRequestLine(this, buffer, out consumed, out examined);
             if (!result && overLength)
             {
-                RejectRequest(RequestRejectionReason.RequestLineTooLong);
+                RequestRejectionUtilities.RejectRequest(RequestRejectionReason.RequestLineTooLong);
```

Comment:
The jit doesn't split functions so will include all the asm at bottom of function to handle the throw. It will be "cold" in that it will be out of flow; but its better to remove the throw itself into void function; when it becomes a single call (that is also moved to bottom of function)